turbo/torrent/postprocessing.go

```diff
@@ -78,24 +77,16 @@ func GenerateHeaderIndexes(ctx context.Context, db ethdb.Database) error {
 
 	if v == 0 {
 		log.Info("Generate headers hash to number index")
-		headHashBytes, innerErr := db.Get(dbutils.SnapshotInfoBucket, []byte(dbutils.SnapshotHeadersHeadHash))
-		if innerErr != nil {
-			return innerErr
-		}
-
-		headNumberBytes, innerErr := db.Get(dbutils.SnapshotInfoBucket, []byte(dbutils.SnapshotHeadersHeadNumber))
-		if innerErr != nil {
-			return innerErr
-		}
 
-		headNumber := big.NewInt(0).SetBytes(headNumberBytes).Uint64()
-		headHash := common.BytesToHash(headHashBytes)
-
-		innerErr = etl.Transform("Torrent post-processing 1", db, dbutils.HeaderPrefix, dbutils.HeaderNumberPrefix, os.TempDir(), func(k []byte, v []byte, next etl.ExtractNextFunc) error {
-			if len(k) != 8+common.HashLength {
+		innerErr := etl.Transform("Torrent post-processing 1", db, dbutils.HeaderPrefix, dbutils.HeaderNumberPrefix, os.TempDir(), func(k []byte, v []byte, next etl.ExtractNextFunc) error {
+			switch {
+			case len(k) == 40:
+				return next(k, common.CopyBytes(k[8:]), common.CopyBytes(k[:8]))
+			case len(k) == 8:
+				return next(k, common.CopyBytes(v[:common.HashLength]), k)
```

Comment:
I suppose it's not correct, because of len(k) in that case = 8. 

I think you can use here just the first case because this part of code is used once to create additional indexes for a downloaded headers snapshot(it contains only canonical headers ).